### PR TITLE
+ Room reservations

### PR DESCRIPTION
**Pull request by theopolisme**
screen shot 2019-02-05 at 9 46 14 pm
screen shot 2019-02-05 at 9 46 02 pm
screen shot 2019-02-05 at 9 45 54 pm

### CODE REVIEW

File: src/Rooms/Rooms.tsx [hunk 1]
```diff
@@ -0,0 +1,148 @@
...
+
+  onReserve(id, e) {
+    e.preventDefault();
+    API.post("treehacks", `/rooms`, { body: { room: id } })
+      .then(this._updateFromApi)
+      .catch(e => alert(`Couldn't reserve room: ${e.response.data.message}`));
+  }
+
+  onCancel() {
+    if (!window.confirm('just making sure—are you sure you\'d like to drop this room? thanks for being so considerate and giving other hacks a chance!')) { return; }
```
Comment (oliviabrown9) [inline src/Rooms/Rooms.tsx]: "hacks" -> "hackers"

**Comment by epicfaace** [comment]
@theopolisme I think we might need to account for time zones. There's a chance people from other countries may not have changed the time on their local computers -- do you think we need to account for that?

File: src/Rooms/Rooms.tsx [hunk 2]
```diff
@@ -0,0 +1,149 @@
...
+          </div>
+        </div>
+      </div>
+    );
+  }
+}
+
+class RoomsWrapper extends React.Component<IRoomsProps, {}> {
+  render() {
+      if (false) {
```
Comment (epicfaace) [inline src/Rooms/Rooms.tsx]: Um I think we can remove this? @theopolisme

File: src/Rooms/Rooms.tsx [hunk 3]
```diff
@@ -0,0 +1,149 @@
...
+            :
+              <div>You have no currently reserved room.</div>
+            }
+          </div>
+
+          <div>
+            <h3>Available rooms</h3>
+            <div className="rooms">
+              {rooms.map(({ id, name, expiry, error }) => {
+                const delta = (+expiry - now) / 1000;
```
Comment (epicfaace) [inline src/Rooms/Rooms.tsx]:
```suggestion
                const delta = (expiry - now) / 1000;
```

**Comment by oliviabrown9** [comment]
@epicfaace I think accounting for time zones is nice to have but maybe not worth the effort

**Comment by theopolisme** [comment]
Where / why would timezones come up? Everything is passed as iso times no?

File: src/Rooms/Rooms.tsx [hunk 4]
```diff
@@ -0,0 +1,149 @@
...
+
+export class Rooms extends React.Component<IRoomsProps, IRoomsState> {
+  _clockInterval: number;
+  _fetchInterval: number;
+
+  constructor(props) {
+    super(props);
+
+    this.state = {
+      now: Date.now(),
```
Comment (epicfaace) [inline src/Rooms/Rooms.tsx]: @theopolisme here is where time zones would come up - `now` is used to calculate `delta`

**Pull request merged**